src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -588,10 +588,6 @@ private void DoStart()
                 }
                 if (logger.IsVerbose) { logger.Verbose("Message gateway service started successfully."); }
 
-                scheduler.QueueTask(clientRegistrar.Start, clientRegistrar.SchedulingContext)
-                    .WaitWithThrow(initTimeout);
```

Comment:
Nevermind, we did not loose the stacktrace, as it was not in a grain call (which is where we actually loose the stacktrace). So yeah, I guess gateway field being null is the only real possibility... not sure how it comes to that, but I assume you found a path